packages/tpetra/core/src/Tpetra_CrsGraph_def.hpp

```diff
@@ -251,6 +251,20 @@ namespace Tpetra {
     , haveGlobalConstants_ (false)
     , sortGhostsAssociatedWithEachProcessor_ (true)
   {
+#ifdef TPETRA_ENABLE_DEPRECATED_CODE
+      #if defined (__GNUC__)
+        #pragma GCC warning "Default profile with deprecated code is DynamicProfile, pftype will soon be unsupported."
```

Comment:
Please don't do this; this will break Sierra's build.  (It won't pass PR tests anyway, since we have a warnings as errors build.)  The right way to do this is to mark the `DynamicProfile` value as deprecated, using the `TPETRA_DEPRECATED` macro.  Check out this example:

https://godbolt.org/z/oJ_ZWk
```c++
#include <iostream>
// Don't actually do this -- Tpetra does it for you.
#define TPETRA_DEPRECATED __attribute ((deprecated))

enum EFoo {
    FOO_ZERO,
    FOO_ONE,
    // This is how you mark an enum value as deprecated.
    FOO_TWO TPETRA_DEPRECATED
};

int main() {
    // This line of code emits a deprecation warning.
    std::cout << FOO_TWO << std::endl;
    return 0;
}
```
Note that if we deprecate `DynamicProfile`, we'll need to change the rest of Tpetra so that it only ever uses the `StaticProfile` enum value.  (Instead of testing `pftype == DynamicProfile`, test `pftype == StaticProfile`.)